libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -751,17 +726,31 @@ class MapboxRouteLineApi(
                 }
             }
 
+        val primaryRouteSource = partitionedRoutes.first.firstOrNull()?.featureCollection
+            ?: FeatureCollection.fromFeatures(
+                listOf()
+            )
+        val alternativeRoute1FeatureCollection =
+            partitionedRoutes.second.firstOrNull()?.featureCollection
+                ?: FeatureCollection.fromFeatures(listOf())
+        val alternativeRoute2FeatureCollection = if (partitionedRoutes.second.size > 1) {
+            partitionedRoutes.second[1].featureCollection
+        } else {
+            FeatureCollection.fromFeatures(listOf())
+        }
+
         val trafficLineExp = trafficLineExpressionDef.await()
         val routeLineExp = routeLineExpressionDef.await()
         val routeCasingLineExp = routeLineCasingExpressionDef.await()
         val altRoute1Exp = alternativeRoute1TrafficExpressionDef.await()
         val altRoute2Exp = alternativeRoute2TrafficExpressionDef.await()
-        val alternativeRoute1FeatureCollection = alternativeRoute1FeatureCollectionDef.await()
-        val alternativeRoute2FeatureCollection = alternativeRoute2FeatureCollectionDef.await()
         val wayPointsFeatureCollection = wayPointsFeatureCollectionDef.await()
-        val primaryRouteSource = primaryRouteSourceDef.await()
         val restrictedSectionFeatureCollection = restrictedSectionFeatureCollectionDef.await()
 
+        partitionedRoutes.first.firstOrNull()?.let {
+            routeLineOptions.vanishingRouteLine?.initWithRoute(it.route)
+        }
```

Comment:
Should we add a test that checks that this operation is scheduled at this point? Trying to avoid a regression.

Reply:
I had the same idea but I can't think of how to test for this.  I added a comment in the code.

Reply:
Maybe using [`verifyOrder`](https://mockk.io/#verification-order)? In any case, we can follow up later. Let's cut a ticket to investigate further.

Reply:
Maybe, I'd have to mock all the methods in the utils class.